### PR TITLE
Added Unversioned Type

### PR DESCRIPTION
**Pull request by Krakaw**
This PR adds an "Unversioned" file type. These files are prefixed with `U` and ignore the order that the migrations get run. As long as the version does not conflict with another migration it doesn't matter what the order is of these files.

### CODE REVIEW

**Review by jxs — CHANGES_REQUESTED** [review]
Hi, and thanks a lot for this! Only a minor issue described below

File: README.md [hunk 1]
```diff
@@ -26,7 +26,8 @@ If you are using a driver that is not yet supported, namely [`SQLx`](https://git
 
 - Add refinery to your Cargo.toml dependencies with the selected driver as feature eg: `refinery = { version = "0.3", features = ["rusqlite"]}`
 - Migrations can be defined in .sql files or Rust modules that must have a function called `migration` that returns a [`String`](https://doc.rust-lang.org/std/string/struct.String.html).
-- Migrations, both .sql files and Rust modules must be named in the format `V{1}__{2}.sql` or `V{1}__{2}.rs`, where `{1}` represents the migration version and `{2}` the name.
+- Migrations can be strictly versioned by prefixing the file with `V` or not strictly versioned by prefixing the file with `U`.
```
[the remark this answers is not in the document]
Comment (Krakaw) [reply README.md:29]: Done!

File: rust-roolchain [hunk 2]
```diff
@@ -0,0 +1 @@
+1.22.1
```
Comment (jxs) [inline rust-roolchain:1]: do we need this?

Comment (Krakaw) [reply rust-roolchain:1]: I don't think so, the `cargo fmt` check was using that version, so I put it in.

**Review by jxs — APPROVED** [review]

**Comment by jxs** [comment]
thanks! :blue_heart:

**Pull request merged**